File: MessageScheduler/MessageScheduler/Models/Message.swift
import Foundation

struct Message: Codable, Identifiable {
    let id: UUID
    var recipient: String
    var content: String
    var scheduledTime: Date
    var isRecurring: Bool
    var recurringInterval: RecurringInterval?
    var isDraft: Bool
    var conditions: MessageConditions?
    
    enum RecurringInterval: String, Codable {
        case daily, weekly, monthly
    }
}

struct MessageConditions: Codable {
    var weekdaysOnly: Bool
    var timeRange: TimeRange?
}

struct TimeRange: Codable {
    var start: Date
    var end: Date
}

struct Contact: Identifiable {
    let id: UUID
    var name: String
    var phoneNumber: String
}

File: app/models/message.ts
export interface Message {
  id: string;
  recipient: string;
  content: string;
  scheduledTime: Date;
  isRecurring: boolean;
  recurringInterval?: 'daily' | 'weekly' | 'monthly';
  isDraft: boolean;
  conditions?: {
    weekdaysOnly: boolean;
    timeRange?: {
      start: string;
      end: string;
    };
  };
}

export interface Contact {
  id: string;
  name: string;
  phoneNumber: string;
}

export interface MessageExport {
  version: string;
  messages: Message[];
  exportDate: string;
}